typescript/trans/typechecking/typescript.nabl2

```diff
@@ -0,0 +1,144 @@
+module typechecking/typescript
+
+imports
+
+  signatures/-
+  typechecking/-
+  
+signature
+    
+    
+    constructors
+    	RECORD	: scope -> Type
+    	
+    	NUMBER	: Type
+		  STRING	: Type
+		  BOOLEAN : Type
+		  FUNCTION: list(Type) * Type -> Type
+		
+
+	name resolution
+		labels
+	      P I
+	
+	    order
+	      D < I,
+	      D < P,
+	      I < P
+	
+	    well-formedness
+	      P* . I*
+      
+		namespaces
+			Field
+			Type
+			Value
+      
+rules
+	
+	init ^ (s) :=
+    	new s.
+    
+    [[ Program(statements) ^ (s) ]] :=
+    	Map1 [[ statements ^ (s) ]],
+    	distinct D(s)/Value | error @ NAMES.
```

Comment:
This one does not work for some reason, but I have no idea why :( The scope graph file does show the correct declarations with the correct constructors, but this line does not error on duplicate values.